projects/RabbitMQ.Client/client/impl/Connection.Receive.cs

```diff
@@ -175,26 +178,29 @@ private void MaybeTerminateMainloopAndStopHeartbeatTimers(bool cancelMainLoop =
             MaybeStopHeartbeatTimers();
         }
 
-        private void HandleMainLoopException(ShutdownEventArgs reason)
+        private Task HandleMainLoopExceptionAsync(ShutdownEventArgs reason)
         {
             string message = reason.GetLogMessage();
             if (false == SetCloseReason(reason))
             {
                 LogCloseError($"Unexpected Main Loop Exception while closing: {message}", reason.Exception);
-                return;
+                return Task.CompletedTask;
             }
 
             _channel0.MaybeSetConnectionStartException(reason.Exception);
 
-            OnShutdown(reason);
             LogCloseError($"Unexpected connection closure: {message}", reason.Exception);
+
+            return OnShutdownAsync(reason);
```

Comment:
missing .ConfigureAwait(false);

Reply:
Same as above ... the `Task` is not awaited here.